The Doc content gate CRASHES on a `$model-probe` hub-activation timeout — no verdict, on unrelated PRs

Two PRs whose diffs cannot reach the Doc content at all — #2467 (one workflow file) and #2461 (CI) — both went red on **Doc content compiles and runs**, with the same shape:

```
warn: MeshWeaver.Mesh.GetMeshNode[0] GetMeshNode('$model-probe/a75e76b9d2fa4ebbaca5e979f648ee2d') timed out
      after 10.0s (budget 10s) — the owning per-node hub never answered the GetDataRequest.
Unhandled exception. System.AggregateException: One or more errors occurred.
 ---> System.TimeoutException: GetMeshNode('$model-probe/…') timed out after 10.0s …
      at System.Reactive.PlatformServices.ExceptionServicesImpl.Rethrow(Exception)
      …
      at MeshWeaver.Mesh.MeshNodeStreamExtensions.<…>g__EmitError|2(Exception)
      at System.Threading.CancellationTokenSource.ExecuteCallbackHandlers(Boolean)
##[error]The content gate failed before it produced a verdict — no check was judged.
```

- #2467 → [job 98407873831](https://github.com/Systemorph/MeshWeaver/actions/runs/33038570991/job/98407873831), probe `a75e76b9…`
- #2461 → [job 98355928593](https://github.com/Systemorph/MeshWeaver/actions/runs/33021741071/job/98355928593), probe `47eef047…`

Same 10 s budget, different probe guid, unrelated diffs. #2456 / #2455 / #2451 passed the same check in the same window, so it is intermittent rather than a trunk break.

## Two defects, and the second is the worse one

**1. A `$model-probe/<guid>` per-node hub does not answer within 10 s.** The message already says this is *not* "node not found" — the hub never answered its `GetDataRequest`. A transient probe address that never activates is a real bug or a budget that is too tight under CI load; either way it is diagnosable from the address family.

**2. The timeout escapes as an UNHANDLED EXCEPTION and takes the process down.** The gate then reports *"failed before it produced a verdict — no check was judged"* — so a run in which the gate learned **nothing** is presented in the same red as a run in which it judged content and found it wrong. Those are different facts and want different responses: the first is "re-run", the second is "fix your content". Worse, the failure is in the family this repo keeps re-learning — a check whose outcome does not mean what its colour says.

A `Rethrow` reaching `ExceptionServicesImpl` from a `ReplaySubject.OnError` means the probe path has no error handler at its subscription seam. The observable is doing the right thing (erroring); nobody is catching it.

## What would close it

- The probe timeout should be an outcome the gate CLASSIFIES, not an exception that ends it — the same reasoning as "a gate that can never open must answer immediately rather than sit on a timeout".
- Distinguish "no verdict" from "verdict: fail" in the annotation, so a reader can tell a flake from a finding without opening the log.
- Then find out why a `$model-probe` hub misses a 10 s activation budget on a loaded runner.

## Escalation: this is now hitting `main`, not only PRs

Run **33085734292** on `main` at `46fde128` (15:14Z):

```
##[error]The content gate failed before it produced a verdict — no check was judged.
FAILED: Doc content compiles and runs
FAILED: Consolidate test results
```

Stack, ending in the same place as this issue's original report:

```
at MeshWeaver.Mesh.MeshNodeStreamExtensions.<>c__DisplayClass6_1.<GetMeshNodeOutcome>g__EmitError|2(Exception)
at System.Reactive.Sink`1.ForwardOnError → Sink`2.OnError → AutoDetachObserver`1.OnErrorCore
at System.Threading.Thread.StartCallback()
```

Also in the same job, and possibly a second defect: `Hub sync/… initialization failed — a BuildupAction …` (compare **#2444**), plus a `ReplaySubject.ReplayBase.OnError` frame.

The commits under it are three merges I made minutes earlier (#2470, #2475, #2484). **None of their diffs touches `Doc/` content or `MeshNodeStreamExtensions`** — #2470 is a workflow + AGENTS.md, #2475 is the combo-verifier and its own tests, #2484 is the recursive-delete fan-out. So this is the known defect firing on main rather than a regression from those merges; recording the attribution here rather than leaving the red unexplained.

## Why the escalation matters beyond one red run

**A gate that dies before judging is worse than a gate that fails.** *"No check was judged"* and *"every check passed"* are one glance apart on a dashboard, and this repo's standing rule is that a gate which cannot report is not a gate. On `main` specifically, the required check going red for a reason unrelated to the tree blocks the merge queue for everyone and trains people to discount it — which is precisely how a real content failure would get waved through.

## The mechanism, stated for whoever picks it up

A `GetMeshNodeOutcome` timeout raises `observer.OnError(...)` from a `cts.Token.Register` callback on a **`TimerQueue` thread**, into a chain terminating at a `Subscribe(onNext)` with **no error arm**. Rx's default handler rethrows on that thread, which is not a test failure but an **unhandled exception that kills the host** — hence no verdict.

Two things need separating, and I have not established which is in play here:

1. **The handling** — a missing error arm. Fixing it converts a host kill into a clean, attributable failure. It does **not** make the probe succeed.
2. **The timeout itself** — `the owning per-node hub never answered the GetDataRequest`. If a `$model-probe` hub never answers, that is its own defect. Prior art worth checking: `5ddaf0d4b` *"A transient probe hub never starts its data sources — the CD-killing flake"* and `a496723b4` splitting `DataContext.Initialize`, both 2026-08-22 — this may be a survivor of that family.

🚨 **Not to be fixed by widening the 10 s budget.** If the hub never answers, the bound is not the problem.

An agent is on this now.

--- a/src/MeshWeaver.Data/TransientNodeProbe.cs
+++ b/src/MeshWeaver.Data/TransientNodeProbe.cs
@@ -19,6 +19,17 @@ namespace MeshWeaver.Data;
 ///
 /// <para>Streams are still created LAZILY when a request actually needs one — the guard removes
 /// only the eager start nobody consumes.</para>
+///
+/// <para>🚨 <b>A probe hub HAS NO MESH NODE, and the read path enforces that.</b> The instance
+/// configuration a probe applies is content written for a REAL per-node hub, where the hub's
+/// address IS its mesh path — so a loader deriving a path from <c>Hub.Address</c> and reading it is
+/// ordinary, correct code that collapses onto the probe's own synthetic address here. Posted to the
+/// probe, such a read parks behind the <c>DataContextInit</c> / <c>MeshNodeInit</c> gates that the
+/// probe's own data-context initialization opens — a CYCLE whose only exit was the read's full
+/// budget, ending on a <c>CancellationTokenSource</c> timer thread (Systemorph/MeshWeaver#2468).
+/// <c>MeshNodeStreamExtensions.GetMeshNodeOutcome</c> therefore answers a probe's read of its OWN
+/// address <c>Absent</c> immediately — the truthful answer, since there is no node there and never
+/// will be. Reads of any REAL path from a probe are untouched.</para>
 /// </summary>
 /// <param name="StartDataSources">Whether the probe still STARTS its data sources on the init
 /// turn. Default TRUE — the pre-existing behaviour, which probes that snapshot actual data (the

--- a/src/MeshWeaver.Data/VirtualDataSource.cs
+++ b/src/MeshWeaver.Data/VirtualDataSource.cs
@@ -112,7 +112,29 @@ public record VirtualDataSource(object Id, IWorkspace Workspace)
                                 return (ChangeItem<EntityStore>?)
                                     new ChangeItem<EntityStore>(newStore, Id.ToString()!, stream.StreamId, ChangeType.Full, stream.Hub.Version, []);
                             }, _ => { });
-                    })
+                    },
+                    // 🚨 THE ERROR ARM IS NOT OPTIONAL — omitting it is a PROCESS KILLER, not a
+                    // missing log line. A virtual type's provider is arbitrary composed content
+                    // (a mesh read, a query hop, a CombineLatest over other streams), so it CAN
+                    // fault; and Rx's default onError handler for a one-argument Subscribe is
+                    // Stubs.Throw, which RETHROWS the fault on whatever thread carried it. That
+                    // thread is almost never one with a catch: in Systemorph/MeshWeaver#2468 the
+                    // provider's `hub.GetMeshNode(...)` timed out, so the OnError originated in a
+                    // CancellationTokenSource callback on a TimerQueue thread — the rethrow became
+                    // an UNHANDLED exception, the host aborted (core dumped), and the Doc content
+                    // gate reported "failed before it produced a verdict — no check was judged".
+                    // A gate that dies before judging is worse than a gate that fails.
+                    //
+                    // Reported, never swallowed: the fault is real and this collection is now
+                    // frozen at its last emission. The data source's own initialization observes
+                    // the SAME faulted Replay(1) (VirtualTypeSource.StreamUpdates), so a fault
+                    // during init still fails the hub's startup through the normal path — this arm
+                    // exists so a fault can never take the process with it.
+                    error => Logger.LogError(error,
+                        "Virtual data source {DataSource}: the provider for collection "
+                        + "'{Collection}' faulted. That collection is frozen at its last emission "
+                        + "and will receive no further updates on hub {Address}",
+                        Id, typeSource.CollectionName, Workspace.Hub.Address))
             );
         }
 

--- a/src/MeshWeaver.Documentation/Data/Architecture/AsynchronousCalls.md
+++ b/src/MeshWeaver.Documentation/Data/Architecture/AsynchronousCalls.md
@@ -315,6 +315,27 @@ pendingFlush.Disposable = Observable.Timer(TimeSpan.FromMilliseconds(100))
 
 Holding it in a field is **not** the same as owning it — the field has to be disposed by the owner's teardown, on every path. Full treatment, with both leak shapes, the measured truth table of which primitives actually root, and why there is deliberately no analyzer: [Subscription Ownership](/Doc/Architecture/SubscriptionOwnership).
 
+### 🚨 …and a subscription on a path that can FAULT needs an error arm — omitting it kills the process
+
+`Subscribe(onNext)` is not "subscribe and ignore errors". Rx's default `onError` for the one-argument overload is `Stubs.Throw`, which **rethrows the fault on whatever thread carried it** — and that thread is chosen by whatever produced the error, not by you. A timeout raises `OnError` from a `CancellationTokenSource` callback on a **`TimerQueue` thread**, where there is no `catch` anywhere above the frame. The result is an unhandled exception and an aborted process, not a logged error.
+
+```csharp
+// ❌ No error arm. A provider timeout here ends the PROCESS on a timer thread.
+typeSource.GetStreamUpdates().Subscribe(instances => Apply(instances));
+
+// ✅ Reported, on the thread the fault arrived on, and the process survives.
+typeSource.GetStreamUpdates().Subscribe(
+    instances => Apply(instances),
+    error => Logger.LogError(error, "…{Collection} faulted; it is frozen at its last emission", name));
+```
+
+Two corollaries worth stating separately:
+
+- **A `try`/`catch` around the `Subscribe` CALL catches nothing.** The fault travels through the stream and arrives later, on another thread; the `try` block has long since exited. Fault handling is fluent — an error arm, or `.Catch(...)` — never `try`.
+- **An error arm is not a swallow.** It must SAY what stopped working. Where another subscriber already owns the verdict (a data source's initialization observing the same `Replay(1)`, which fails the hub's startup), the arm's job is to report and let that path decide — its existence is what stops a fault from taking the host with it.
+
+This is the defect behind [#2468](https://github.com/Systemorph/MeshWeaver/issues/2468): a virtual data source's live-update subscription had no error arm, so a content loader's `GetMeshNode` timeout aborted the CI content gate — which then reported *"failed before it produced a verdict — no check was judged"*. A gate that dies before judging is worse than a gate that fails.
+
 ---
 
 ## 🚨 `MeshNode.Content` is always typed at the `GetMeshNodeStream` boundary

--- a/src/MeshWeaver.Documentation/Data/WhatsNew/2026-08-27-content-gate-no-verdict.md
+++ b/src/MeshWeaver.Documentation/Data/WhatsNew/2026-08-27-content-gate-no-verdict.md
@@ -1,0 +1,19 @@
+---
+Name: Data views survive a failing data provider
+Category: Fix
+Description: A data source whose provider fails now reports the failure instead of taking the whole process down.
+Icon: Sparkle
+Order: -20260827
+---
+
+# Data views survive a failing data provider
+
+A view backed by computed ("virtual") data could take the whole application down when the
+computation behind it failed — for example when a lookup it needed timed out. The failure was
+re-thrown on a background timer, where nothing could catch it, so instead of one view showing an
+error the process ended abruptly and whatever it was doing produced no result at all.
+
+That failure is now reported: the affected collection stops updating and says so in the log, and
+everything else keeps running. A related cause is fixed too — the short-lived hub used to work out
+a node type's data model no longer stalls for ten seconds when the type's own code looks up the
+node it is running on.

--- a/src/MeshWeaver.Mesh.Contract/MeshNodeStreamExtensions.cs
+++ b/src/MeshWeaver.Mesh.Contract/MeshNodeStreamExtensions.cs
@@ -2145,8 +2145,10 @@ public static class MeshNodeStreamExtensions
     /// <list type="bullet">
     ///   <item><see cref="NodeReadStatus.Present"/> — a node came back.</item>
     ///   <item><see cref="NodeReadStatus.Absent"/> — routing said
-    ///     <see cref="ErrorType.NotFound"/>, the owner answered with no data, or a read validator
-    ///     hid it (a hidden node is invisible by contract).</item>
+    ///     <see cref="ErrorType.NotFound"/>, the owner answered with no data, a read validator
+    ///     hid it (a hidden node is invisible by contract), or the reader is a TRANSIENT NODE
+    ///     PROBE being asked for its OWN address, which by contract is never a mesh node — that
+    ///     one is answered without a round-trip at all (see the guard in the body).</item>
     ///   <item><see cref="NodeReadStatus.DeleteInProgress"/> — the owner answered with its delete
     ///     tombstone (<c>GetDataResponse.Absence</c>).</item>
     ///   <item><see cref="NodeReadStatus.Unavailable"/> — the delivery failed for any other
@@ -2180,6 +2182,51 @@ public static class MeshNodeStreamExtensions
             // "DeliveryFailure … (sender: Plugins/_DefaultInstallLedger)"). Same seam MeshService
             // uses for CRUD; a non-router caller gets itself back, unchanged.
             var issuingHub = hub.NodeOperationIssuingHub();
+
+            // ♻️ A TRANSIENT NODE PROBE HAS NO MESH NODE — so reading its OWN address is a CYCLE,
+            // and the only way it ever ended was by spending the entire budget.
+            //
+            // A probe hub (`$model-probe/{guid}`, `$schema-probe/{guid}`) exists to have a
+            // NodeType's INSTANCE configuration applied to it so its type registry / schema can be
+            // snapshotted, and is disposed in the same breath. AsTransientNodeProbe states the
+            // contract outright: "with no own-node subscription and no persistence sampler it has
+            // no node identity". But that instance configuration is content written for a REAL
+            // per-node hub, where the hub's address IS its mesh path — so deriving a path from
+            // `Hub.Address` and reading it is an ordinary, correct thing for a loader to do. On the
+            // probe that same derivation collapses onto the probe's own synthetic address.
+            //
+            // The read is then posted to the probe itself, where it parks behind the
+            // DataContextInit / MeshNodeInit gates — and it is the probe's OWN data-context
+            // initialization that issued it, so the gates cannot open until the read completes and
+            // the read cannot complete until the gates open. Systemorph/MeshWeaver#2468: every such
+            // read burned its full 10 s and then errored from the CTS timer thread.
+            //
+            // Absent is the TRUTHFUL answer, not a shortcut: there is no node at this address and
+            // there never will be. Answering immediately is the same reasoning as "a gate that can
+            // never open must answer immediately rather than sit on a timeout" — and it is scoped
+            // to the probe's own address, so a probe reading any REAL path is untouched.
+            if (issuingHub.Configuration.Get<TransientNodeProbe>() is not null
+                && string.Equals(path, issuingHub.Address.ToString(), StringComparison.Ordinal))
+            {
+                try
+                {
+                    issuingHub.ServiceProvider.GetService<ILoggerFactory>()
+                        ?.CreateLogger("MeshWeaver.Mesh.GetMeshNode")
+                        ?.LogDebug(
+                            "GetMeshNode('{Path}') on a transient node probe reads the probe's OWN "
+                            + "address — a probe has no mesh node, so this is answered Absent "
+                            + "immediately rather than parked behind the probe's own init gates.",
+                            path);
+                }
+                catch
+                {
+                    // Logging must never mask the verdict it is reporting.
+                }
+                observer.OnNext(NodeReadOutcome.Absent);
+                observer.OnCompleted();
+                return Disposable.Empty;
+            }
+
             var budget = timeout ?? TimeSpan.FromSeconds(10);
             var started = Stopwatch.StartNew();
             var cts = new CancellationTokenSource(budget);

--- a/test/MeshWeaver.Data.Test/VirtualDataSourceProviderFaultTest.cs
+++ b/test/MeshWeaver.Data.Test/VirtualDataSourceProviderFaultTest.cs
@@ -1,0 +1,75 @@
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Reactive.Subjects;
+using System.Threading.Tasks;
+using MeshWeaver.Fixture;
+using MeshWeaver.Messaging;
+using Xunit;
+
+namespace MeshWeaver.Data.Test;
+
+/// <summary>A row produced by the faulting virtual provider below.</summary>
+/// <param name="Id">Key.</param>
+/// <param name="Name">Payload.</param>
+public record FaultingRow([property: Key] int Id, string Name);
+
+/// <summary>
+/// Pins the ERROR ARM on <c>VirtualDataSource</c>'s live-update subscription
+/// (Systemorph/MeshWeaver#2468).
+///
+/// <para>A virtual type's provider is arbitrary composed content — a mesh read, a query hop, a
+/// <c>CombineLatest</c> over other streams — so it CAN fault. The subscription that pushes its
+/// later emissions into the data source used to be a one-argument <c>Subscribe(onNext)</c>, and
+/// Rx's default <c>onError</c> for that overload is <c>Stubs.Throw</c>: it RETHROWS the fault on
+/// whatever thread carried it.</para>
+///
+/// <para>That thread is almost never one with a catch. In #2468 the provider's
+/// <c>hub.GetMeshNode(...)</c> timed out, so the <c>OnError</c> originated inside a
+/// <c>CancellationTokenSource</c> callback on a <c>TimerQueue</c> thread; the rethrow was an
+/// UNHANDLED exception, the host aborted (core dumped), and the Doc content gate reported
+/// <i>"failed before it produced a verdict — no check was judged"</i>. A gate that dies before
+/// judging is worse than a gate that fails, so this is pinned rather than left to review.</para>
+///
+/// <para>The test drives the fault from the TEST thread precisely because the rethrow is
+/// synchronous on the emitting thread — that is what makes it observable here at all, and it is
+/// the same rethrow that has no catch in production.</para>
+/// </summary>
+public class VirtualDataSourceProviderFaultTest(ITestOutputHelper output) : HubTestBase(output)
+{
+    private readonly BehaviorSubject<IEnumerable<FaultingRow>> rows =
+        new([new FaultingRow(1, "seed")]);
+
+    /// <inheritdoc />
+    protected override MessageHubConfiguration ConfigureHost(MessageHubConfiguration configuration)
+        => base.ConfigureHost(configuration)
+            .AddData(data => data
+                .WithVirtualDataSource("Faulting", vds =>
+                    vds.WithVirtualType<FaultingRow>(_ => rows)));
+
+    /// <summary>
+    /// A faulting provider must be REPORTED by the data source, never rethrown onto the thread
+    /// that produced the fault.
+    /// </summary>
+    [Fact(Timeout = 30000)]
+    public async Task ProviderFault_IsReported_NeverRethrownOnTheEmittingThread()
+    {
+        var workspace = GetHost().GetWorkspace();
+
+        // Let the source come up first, so the live-update subscription — the one under test — is
+        // the subscriber that is still attached when the fault lands (initialization's Take(1) has
+        // already released).
+        var stream = workspace.GetStream(typeof(FaultingRow));
+        await stream.Should().Within(TimeSpan.FromSeconds(10)).Emit();
+
+        var fault = new TimeoutException(
+            "GetMeshNode('$model-probe/deadbeef') timed out after 10.0s — the shape of #2468");
+
+        var act = () => rows.OnError(fault);
+
+        act.Should().NotThrow(
+            "a virtual provider's fault must be reported by the data source, not rethrown onto "
+            + "the emitting thread — in production that thread is a CTS TimerQueue thread with "
+            + "no catch anywhere above it, so the rethrow aborts the process");
+    }
+}

--- a/test/MeshWeaver.Graph.Test/NodeTypeModelProbeTest.cs
+++ b/test/MeshWeaver.Graph.Test/NodeTypeModelProbeTest.cs
@@ -3,6 +3,7 @@
 using System;
 using System.Collections.Concurrent;
 using System.Collections.Generic;
+using System.Diagnostics;
 using System.Linq;
 using System.Reactive.Disposables;
 using System.Reactive.Linq;
@@ -55,9 +56,28 @@ public class NodeTypeModelProbeTest(ITestOutputHelper output) : MonolithMeshTest
         => base.ConfigureMesh(builder)
             .ConfigureServices(services => services.AddSingleton<ILoggerProvider>(recorder));
 
+    public record ProbeSelfRead(string Id, string Verdict);
+
     /// <summary>The instance configuration a NodeType would apply to its instances.</summary>
     private static MessageHubConfiguration InstanceConfig(MessageHubConfiguration config)
         => config.AddMeshDataSource(source => source.WithContentType<ProbeOrder>());
+
+    /// <summary>
+    /// The same instance configuration plus what real NodeType content routinely does: derive a
+    /// mesh path from the hub's OWN address and read it. On a real per-node hub the address IS the
+    /// node's path, so this is correct code; on the probe it collapses onto the synthetic
+    /// <c>$model-probe/{guid}</c>. See the test below for why that used to cost 10 s and a process.
+    /// </summary>
+    private static MessageHubConfiguration SelfReadingInstanceConfig(MessageHubConfiguration config)
+        => config
+            .AddMeshDataSource(source => source.WithContentType<ProbeOrder>())
+            .AddData(data => data
+                .WithVirtualDataSource("SelfRead", vds =>
+                    vds.WithVirtualType<ProbeSelfRead>(workspace =>
+                        workspace.Hub
+                            .GetMeshNode(workspace.Hub.Address.ToString(), TimeSpan.FromSeconds(10))
+                            .Select(node => (IEnumerable<ProbeSelfRead>)
+                                [new ProbeSelfRead("1", node?.Path ?? "absent")]))));
 
     [Fact(Timeout = 60000)]
     public async Task ProbeInstanceModel_SnapshotsContentTypeAndSchema()
@@ -140,6 +160,40 @@ public class NodeTypeModelProbeTest(ITestOutputHelper output) : MonolithMeshTest
             "a probe hub carries no per-node control plane, so disposing it faults nothing");
         underProbe.Count.Should().BeLessThanOrEqualTo(5,
             "the probe needs its data context, not a sync/ sub-hub per NodeType watcher");
+    }
+
+    /// <summary>
+    /// A probe hub has NO MESH NODE — so a read of its own address must be answered immediately,
+    /// not parked behind the probe's own initialization gates (Systemorph/MeshWeaver#2468).
+    ///
+    /// <para>The read is issued by the probe's own data-context initialization and posted to the
+    /// probe itself, where it defers behind <c>DataContextInit</c> / <c>MeshNodeInit</c> — the very
+    /// gates that initialization opens. A cycle: the read could only ever end by spending its whole
+    /// budget. In CI that budget elapsed on a <c>CancellationTokenSource</c> timer thread, which is
+    /// how a <b>content</b> loader's mesh read ended up aborting the Doc content gate's process.</para>
+    ///
+    /// <para>The bar is elapsed time, deliberately: the assertion has to fail if the read parks,
+    /// and 8 s is under the read's own 10 s budget while leaving a loaded runner room for the
+    /// probe's ordinary cost (~1 s locally).</para>
+    /// </summary>
+    [Fact(Timeout = 60000)]
+    public async Task ProbeInstanceModel_OwnAddressRead_IsAnsweredWithoutBurningTheBudget()
+    {
+        var started = Stopwatch.StartNew();
+        var model = await NodeTypeDataModelAreas
+            .ProbeInstanceModel(Mesh, SelfReadingInstanceConfig)
+            .Should().Within(40.Seconds()).Emit();
+        started.Stop();
+
+        Output.WriteLine($"probe with a self-reading provider completed in {started.Elapsed}");
+
+        model.Should().NotBeNull(
+            "content that reads its own address must not stop the probe from snapshotting");
+        model!.ContentTypeName.Should().Be(nameof(ProbeOrder));
+
+        started.Elapsed.Should().BeLessThan(TimeSpan.FromSeconds(8),
+            "a read of the probe's own address is answered Absent immediately — parking it behind "
+            + "the probe's own init gates is a cycle that can only end by burning the full budget");
     }
 
     private bool ProbeStillLive(string address)

